### PR TITLE
Add more error subcodes to Kuzzle

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

This PR adds more subcodes into kuzzle core
This should be the last PR about adding error subcodes

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :  
  ...

### CODE REVIEW

File: test/api/core/pluginsManager/init.test.js [hunk 1]
```diff
@@ -341,7 +341,7 @@ describe('PluginsManager', () => {
       mockrequire('/kuzzle/plugins/available/kuzzle-plugin-test', pluginStub);
       mockrequire('/kuzzle/plugins/available/kuzzle-plugin-test/manifest.json', {
         name: 'kuzzle-plugin-test',
-        kuzzleVersion: '^2.x'
+        kuzzleVersion: '^1.x'
```
Comment (benoitvidis) [inline test/api/core/pluginsManager/init.test.js]:
```suggestion
        kuzzleVersion: '^2.x'
```

File: test/api/core/pluginsManager/init.test.js [hunk 2]
```diff
@@ -355,7 +355,7 @@ describe('PluginsManager', () => {
         .match({
           name: instanceName,
           privileged: false,
-          kuzzleVersion: '^2.x',
+          kuzzleVersion: '^1.x',
```
Comment (benoitvidis) [inline test/api/core/pluginsManager/init.test.js]:
```suggestion
          kuzzleVersion: '^2.x',
```

File: test/api/core/httpRouter/httpRouter.test.js [hunk 3]
```diff
@@ -23,9 +23,13 @@ describe('core/httpRouter', () => {
...
-    rq = new HttpMessage(
-      { id: 'requestId' },
-      { url: '', method: '', headers: {} });
+    rq = {
+      requestId: 'requestId',
+      url: '',
+      method: '',
+      headers: {},
+      content: ''
+    };
```
Comment (benoitvidis) [inline test/api/core/httpRouter/httpRouter.test.js]: Looks you have reverted this file updates.

File: package.json [hunk 4]
```diff
@@ -22,7 +22,8 @@
     "doc-upload": "cd doc && bash doc.sh upload",
     "doc-cloudfront": "cd doc && bash doc.sh cloudfront",
     "doc-deploy": "npm run doc-upload && npm run doc-cloudfront",
-    "doc-netlify": "npm run doc-prepare && cd doc && bash doc.sh build-netlify"
+    "doc-netlify": "npm run doc-prepare && cd doc && bash doc.sh build-netlify",
+    "doc-generate-subcodes": "cd doc && node generate-subcodes-doc"
```
Comment (benoitvidis) [inline package.json]: Is there a real need to depend on https://github.com/kuzzleio/kuzzle/pull/1405 ?

File: test/api/core/entrypoints/protocols/websocket.test.js [hunk 5]
```diff
@@ -651,6 +651,50 @@ describe('/lib/api/core/entrypoints/protocols/websocket', () => {
       should(activeConnection.socket.terminate).not.be.called();
       should(activeConnection.socket.ping).not.be.called();
     });
+
+    it('should mark a socket as dead if not available for PING', () => {
```
Comment (benoitvidis) [inline test/api/core/entrypoints/protocols/websocket.test.js]: Comes from https://github.com/kuzzleio/kuzzle/pull/1408. Do we need to depend on it?

**Comment by Aschen** [comment]
@Yoann-Abbes You should merge `2-dev` in your PR to make the tests pass

**Comment by scottinet** [comment]
I'm redoing the error codes, so this PR is now obsolete.

**Pull request closed without merging**